docs/book/src/forc/manifest_reference.md

```diff
@@ -41,8 +42,64 @@ entry = "main.sw"
 organization = "Fuel_Labs"
 license = "Apache-2.0"
 name = "wallet_contract"
+
+[project.metadata]
+indexing = { namespace = "counter-contract", schema_path = "out/release/counter-contract-abi.json" }
 ```
 
+### Metadata Section in Forc.toml
+
+The `[project.metadata]` section provides a dedicated space for external tools and plugins to store their configuration in `Forc.toml`. The metadata key names are arbitrary and do not need to match the tool's name.
+
+#### Usage
+
+Use any descriptive key name for your metadata table:
+
+```toml
+[project.metadata.any_name_here]
+option1 = "value"
+option2 = "value"
+
+[project.metadata.my_custom_config]
+setting1 = "value"
+setting2 = "value"
+```
+
+Example from an indexing tool:
+
+```toml
+[project.metadata.indexing]
+namespace = "counter-contract"
+schema_path = "out/release/counter-contract-abi.json"
+```
+
+#### Guidelines for Plugin Developers
+
+Best Practices
+
+* Choose clear, descriptive metadata key names
+* Document the exact metadata key name your tool expects
+* Don't require `Forc.toml` if tool can function without it
+* Consider using TOML format for dedicated config files
+
+Implementation Notes
+
+* The metadata section is optional
+* Forc does not parse metadata contents
+* Plugin developers handle their own configuration parsing
```

Comment:
nit: It would be good to link to the example of metadata parsing here.

Reply:
Nice, it would also be useful to describe workspace and project level metadata and how these are achieved.

Reply:
addressed: https://github.com/FuelLabs/sway/pull/6728/commits/3f05c0628257ba7ee5e71698eb2c0e2772ae3d6e